### PR TITLE
add_depth_estimation_tutorial

### PR DESCRIPTION
**Pull request by isayoften**
New Chapter "Metric and Relative Monocular Depth Estimation: An Overview. Fine-Tuning Depth Anything V2"
Approval: https://github.com/johko/computer-vision-course/discussions/321

### CODE REVIEW

**Comment by johko** [comment]
Hey @isayoften, thank you for that very extensive contribution, just wanted to let you know that I'm a bit short on time currently, but I will review as soon as possible

**Comment by isayoften** [comment]
Hey, is everything okay?

**Comment by johko** [comment]
> Hey @johko, is everything okay?

Hey, yeah sorry I was a little busy and at a conference in Africa, but I also thought about the PR and that I have not fully reviewed it yet. Sorry for the long wait. I'll do it asap

File: chapters/en/unit8/monocular_depth_estimation.mdx [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+
+We'll go through everything in detail here. To start, let's download all the necessary modules from the authors' repository and the checkpoint of the smallest model with the ViT-S encoder.
+
+#### Step 1: Clone the Repository and Download Pre-trained Weights
+
+```bash
+!git clone https://github.com/DepthAnything/Depth-Anything-V2
+!wget -O depth_anything_v2_vits.pth https://huggingface.co/depth-anything/Depth-Anything-V2-Small/resolve/main/depth_anything_v2_vits.pth?download=true
+```
+If you don't work at Kaggle, you can download the dataset [here](http://datasets.lids.mit.edu/fastdepth/data/)
```
Comment (johko) [inline chapters/en/unit8/monocular_depth_estimation.mdx:153]: Do you mean "registered at Kaggle" or really "work at Kaggle"?
```suggestion
If you're not registered at Kaggle, you can download the dataset [here](http://datasets.lids.mit.edu/fastdepth/data/) 
```

**Review by johko — COMMENTED** [review]
So sorry for the long delay, I sped up the reviewing a bit from about half-way and also did not check the code thoroughly. But I think it should definitely be good enough ;)

I really love the content, especially with so many depth models coming out these days, this is a really elaborate section to get started and up-to-date with the whole topic, thank you so much :heart: 

Just a few little nits from my side ;)

File: chapters/en/unit8/monocular_depth_estimation.mdx [hunk 2]
```diff
@@ -0,0 +1,546 @@
...
+
+Typically, the architecture for this image-to-image task is an encoder-decoder model, like a U-Net, with various modifications. Formally, this is a pixel-wise regression problem. Imagine how challenging it is for a neural network to accurately predict distances for each pixel, ranging from a few meters to several hundred meters. <br> This brings us to the idea of moving away from a universal model that predicts exact distances in all scenarios. Instead, let's develop a model that approximately (relatively) predicts depth, capturing the shape and structure of the scene by indicating which objects are farther and which are closer relative to each other and to us. If precise distances are needed, we can fine-tune this relative model on a specific dataset, leveraging its existing understanding of the task.
+
+![image/png](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/Metric%20and%20Relative%20Monocular%20Depth%20Estimation%20An%20Overview.%20Fine-Tuning%20Depth%20Anything%20V2/different_scales.png)
+
+The problems don't end there
```
[the remark this answers is not in the document]
Comment (isayoften) [reply chapters/en/unit8/monocular_depth_estimation.mdx:52]: Agreed

File: chapters/en/unit8/monocular_depth_estimation.mdx [hunk 3]
```diff
@@ -0,0 +1,546 @@
...
+
+Notice how changes in focal length dramatically alter the perception of background distances!
+
+Lastly, many datasets lack absolute depth maps altogether and only have relative ones (for instance, due to the lack of camera calibration). Additionally, each method of obtaining depth has its own advantages, disadvantages, biases, and problems.
+
+![image/png](https://huggingface.co/datasets/hf-vision/course-assets/resolve/main/Metric%20and%20Relative%20Monocular%20Depth%20Estimation%20An%20Overview.%20Fine-Tuning%20Depth%20Anything%20V2/metrics1.png)
+
+>We identify three major challenges. 1) Inherently different representations of depth: direct vs. inverse depth representations. 2) Scale ambiguity: for some data sources, depth is only given up to an unknown scale. 3) Shift ambiguity: some datasets provide disparity only up to an unknown scale and global disparity shift that is a function of the unknown baseline and a horizontal shift of the principal points due to post-processing
```
[the remark this answers is not in the document]
Comment (isayoften) [reply chapters/en/unit8/monocular_depth_estimation.mdx:66]: This is a quote from the article. 
I'll add about disparity.

File: chapters/en/unit8/monocular_depth_estimation.mdx [hunk 4]
```diff
@@ -0,0 +1,546 @@
...
+where \\( d_i^* \\) and \\( d_i \\) are the prediction and ground truth, respectively. And \\( \rho \\) is the affine-invariant mean absolute error loss: \\( \rho(d_i^*, d_i) = \left| \hat{d}_i^* - \hat{d}_i \right| \\), where \\( \hat{d}_i^* \\) and \\( \hat{d}_i \\) are the scaled and shifted versions of the prediction \\( d_i^* \\) and ground truth \\( d_i \\):
+$$\hat{d}_i = \frac{d_i - t(d)}{s(d)},$$
+where \\( t(d) \\) and \\( s(d) \\) are used to align the prediction and ground truth to have zero translation and unit scale:
+$$t(d) = \mathrm{median}(d), \quad s(d) = \frac{1}{HW} \sum_{i=1}^{HW} \left| d_i - t(d) \right|.$$
+
+In fact, there are many other methods and functions that help eliminate scale and shifts. There are also different additions to loss functions, such as gradient loss, which focuses not on the pixel values themselves but on how quickly they change (hence the name — gradient). You can read more about this in the MiDaS paper, i'll include a list of useful literature at the end. Let's briefly discuss metrics before moving on to the most exciting part — fine-tuning on absolute depth using a custom dataset.
```
[the remark this answers is not in the document]
Comment (isayoften) [reply chapters/en/unit8/monocular_depth_estimation.mdx:83]: Agreed

File: chapters/en/unit8/monocular_depth_estimation.mdx (hunk 1 above)
Comment (isayoften) [reply chapters/en/unit8/monocular_depth_estimation.mdx:153]: I'll just rewrite it more logically, leaving only “you can also download the dataset here”

**Review by isayoften — COMMENTED** [review]
I'll make the changes and correct it now.

**Review by johko — APPROVED** [review]
LGTM :+1:

**Comment by isayoften** [comment]
Thanks! So, now i have to wait for the response from the merveenoyan, right?

**Comment by johko** [comment]
> Thanks! So, now i have to wait for the response from the merveenoyan, right?

Not exactly, we just need another review from any other person that is officially a contributor to the repo. I already mentioned the PR in our discord channel, so I hope it won't take long for someone to check and give feedback.

But if that takes too long, I can also just merge it as is (but that is just for "emergency" cases) :wink:

**Review by ATaylorAerospace — APPROVED** [review]
@isayoften this is really informative and detailed and I definitely like the strong use of citations to create the tutorial

**Comment by ATaylorAerospace** [comment]
Merging!

**Pull request merged**